### PR TITLE
linux-cachyos: init at 6.2.10

### PR DESCRIPTION
**Pull request by dr460nf1r3**
Having to pre-compile is tedious without proper hardware.

### CODE REVIEW

File: pkgs/linux-cachyos/config.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  HZ_500 = yes;
+
+  # Disable MQ Deadline I/O scheduler
+  MQ_IOSCHED_DEADLINE = lib.mkForce no;
+
+  # Disable Kyber I/O scheduler
+  MQ_IOSCHED_KYBER = lib.mkForce no;
+
+  # Enabling full ticks
+  CONTEXT_TRACKING = yes;
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:21]: This is the upstream value.
```suggestion
```

File: pkgs/linux-cachyos/config.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  CONTEXT_TRACKING_FORCE = option no;
+  HZ_PERIODIC = no;
+  NO_HZ = yes;
+  NO_HZ_COMMON = yes;
+  NO_HZ_FULL = yes;
+  NO_HZ_FULL_NODEF = option yes;
+  NO_HZ_IDLE = no;
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:28]: The others are the upstream value.
```suggestion
  HZ_PERIODIC = no;
  NO_HZ_FULL_NODEF = option yes;
  NO_HZ_IDLE = no;
```

File: pkgs/linux-cachyos/config.nix [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  LOCALVERSION = freeform "-cachyos-bore";
+  EXPERT = yes;
+  WERROR = no;
+
+  # Bore scheduler
+  SCHED_BORE = yes;
+
+  # Tick to 750hz
+  HZ = freeform "500";
+  HZ_500 = yes;
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:12]:
```suggestion
  HZ_500 = yes;
  HZ_1000 = no;
```

We have to remove the upstream value.

File: pkgs/linux-cachyos/config.nix [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  DEFAULT_BBR2 = yes;
+  DEFAULT_CUBIC = option no;
+  DEFAULT_TCP_CONG = freeform "bbr2";
+  TCP_CONG_BBR2 = yes;
+  TCP_CONG_CUBIC = lib.mkForce module;
+
+  # LRU config
+  LRU_GEN = yes;
+  LRU_GEN_ENABLED = yes;
+  LRU_GEN_STATS = no;
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:44]: These are the upstream values.
```suggestion
  # LRU config
  LRU_GEN = yes;
  LRU_GEN_ENABLED = yes;
  LRU_GEN_STATS = no;
```

File: pkgs/linux-cachyos/config.nix [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  ZRAM_DEF_COMP = freeform "zstd";
+  ZRAM_DEF_COMP_LZORLE = no;
+  ZRAM_DEF_COMP_ZSTD = yes;
+  ZSTD_COMPRESSION_LEVEL = freeform "19";
+  ZSWAP_COMPRESSOR_DEFAULT = freeform "zstd";
+  ZSWAP_COMPRESSOR_DEFAULT_LZ4 = no;
+  ZSWAP_COMPRESSOR_DEFAULT_ZSTD = yes;
+
+  # Enable USER_NS_UNPRIVILEGED
+  USER_NS = yes;
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:58]: This is the upstream value.
```suggestion
```

File: pkgs/linux-cachyos/config.nix [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+  # Enable zram/zswap ZSTD compression
+  MODULE_COMPRESS_ZSTD_LEVEL = option (freeform "9");
+  MODULE_COMPRESS_ZSTD_ULTRA = option no;
+  ZRAM_DEF_COMP = freeform "zstd";
+  ZRAM_DEF_COMP_LZORLE = no;
+  ZRAM_DEF_COMP_ZSTD = yes;
+  ZSTD_COMPRESSION_LEVEL = freeform "19";
+  ZSWAP_COMPRESSOR_DEFAULT = freeform "zstd";
+  ZSWAP_COMPRESSOR_DEFAULT_LZ4 = no;
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:54]:
```suggestion
  ZSWAP_COMPRESSOR_DEFAULT_LZO = no;
```
In NixOS, the default is LZO and not LZ4.

File: pkgs/linux-cachyos/config.nix [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  # # ZRAM & Zswap
+  # Z3FOLD = no;
+  # ZBUD = lib.mkForce no;
+  # ZRAM = module;
+  # ZRAM_DEF_COMP_ZSTD = yes;
+  # ZSMALLOC = lib.mkForce yes;
+  # ZSWAP_COMPRESSOR_DEFAULT_ZSTD = yes;
+  # ZSWAP_DEFAULT_ON = yes;
+  # ZSWAP_ZPOOL_DEFAULT_ZSMALLOC = yes;
+
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:73]:
```suggestion
```
No need for commented configs in-repo.

File: pkgs/linux-cachyos/config.nix [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  # Haswell & newer
+  GENERIC_CPU3 = yes;
+
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:76]:
```suggestion
```
Sorry, nixpkgs supports everything down to x86-64-v1

File: pkgs/linux-cachyos/config.nix [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  # AMD P-state driver
+  # Could cause issues in AMD Virtual machine
+  X86_AMD_PSTATE = yes;
+
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:80]:
```suggestion
```
This is the upstream value.

File: pkgs/linux-cachyos/config.nix [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  # Paragon's NTFS3 driver
+  NTFS3_FS = module;
+  NTFS3_FS_POSIX_ACL = yes;
+  NTFS3_LZX_XPRESS = yes;
+
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:85]:
```suggestion
```

This is the upstream value.

File: pkgs/linux-cachyos/config.nix [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+  PREEMPT_RCU = yes;
+  TASKS_RCU = yes;
+  UNINLINE_SPIN_UNLOCK = yes;
```
Comment (PedroHLC) [inline pkgs/linux-cachyos/config.nix:88]:
```suggestion
```

These are the upstream value.

**Review by PedroHLC — CHANGES_REQUESTED** [review]

**Review by ptr1337 — APPROVED** [review]
Seems good to me.

Hopefully ZFS support is now working proper.
Thanks!

**Review by PedroHLC — APPROVED** [review]
Probably needs some tweaking, but I tested it here with:
```nix
{
  # Creates a second boot entry for testing cachyos kernels
  specialisation.ptr1337.configuration = {
    system.nixos.tags = [ "ptr1337" ];
    boot.kernelPackages = lib.mkForce pkgs.linuxPackages_cachyos;
    boot.zfs.enableUnstable = true;
  };
}
```

and I can see that it worked:
```console
╭─pedrohlc at desktop in /home/pedrohlc
╰─λ uname -a
Linux desktop 6.2.10 #1-NixOS SMP PREEMPT_DYNAMIC Thu Apr  6 10:12:48 UTC 2023 x86_64 GNU/Linux
╭─pedrohlc at desktop in /home/pedrohlc
╰─λ zgrep 'BORE' /proc/config.gz
CONFIG_SCHED_BORE=y
```

[config.gz](https://github.com/chaotic-cx/nyx/files/11204284/config.gz)

**Comment by ptr1337** [comment]
IS the config now fixed? Does it work now correctly?

**Comment by dr460nf1r3** [comment]
It *should* - I've provided the config in the group for you to check a few days ago 👀

**Pull request merged**